CMakeLists.txt

```diff
@@ -324,20 +332,39 @@ endif()
 # ==================================================================================================
 # Functions
 # ==================================================================================================
-function(list_licenses OUTPUT MODULES)
-    set(CONTENT)
-    set(_MODULES ${MODULES} ${ARGN})
-    foreach(module ${_MODULES})
-        set(license_path "../../third_party/${module}/LICENSE")
-        get_filename_component(fullname "${license_path}" ABSOLUTE)
-        string(APPEND CONTENT "License and copyrights for ${module}:\n\n")
-        file(READ ${license_path} license)
-        string(APPEND CONTENT ${license})
-        string(APPEND CONTENT "\n\n")
-    endforeach()
-    configure_file(${FILAMENT}/build/licenses.inc.in ${OUTPUT})
-endfunction(list_licenses)
-
+if (MSVC)
```

Comment:
Do we need two separate code paths here? Or could we use your "multiple string literal" version across all compilers?

Reply:
You are right, we could use my new version across all compilers. Indeed this is what I had done first, but in the meanwhile @prideout made a change in the same [function](https://github.com/google/filament/commit/394db90ec97ed8f9315689529679650935a7a407#diff-af3b638bc2a3e6c650974192a53c7291) which I didn't quite understand the motivation for, so I was hesitant to change it. But you are right it would be much better to merge Philip's change and mine.  

The idea of my change is that, instead of creating a single very long string litteral of the form:

```
R"FILAMENT__(very long string)FILAMENT__"
```

I generate something like:

```
$ cat ./tools/filamesh/generated/licenses/licenses.inc
 R"FILAMENT__()FILAMENT__",
R"FILAMENT__(License and copyrights for getopt:

)FILAMENT__",
R"FILAMENT__(Copyright (c) 1987, 1993, 1994, 1995)FILAMENT__",
R"FILAMENT__(   The Regents of the University of California.  All rights reserved.)FILAMENT__",
R"FILAMENT__()FILAMENT__",
R"FILAMENT__(Redistribution and use in source and binary forms, with or without)FILAMENT__",
R"FILAMENT__(modification, are permitted provided that the following conditions)FILAMENT__",
R"FILAMENT__(are met:)FILAMENT__",
R"FILAMENT__(1. Redistributions of source code must retain the above copyright)FILAMENT__",
R"FILAMENT__(   notice, this list of conditions and the following disclaimer.)FILAMENT__",
R"FILAMENT__(2. Redistributions in binary form must reproduce the above copyright)FILAMENT__",
R"FILAMENT__(   notice, this list of conditions and the following disclaimer in the)FILAMENT__",
R"FILAMENT__(   documentation and/or other materials provided with the distribution.)FILAMENT__",
R"FILAMENT__(3. Neither the names of the copyright holders nor the names of its)FILAMENT__",
R"FILAMENT__(   contributors may be used to endorse or promote products derived from)FILAMENT__",
R"FILAMENT__(   this software without specific prior written permission.)FILAMENT__",
```

Reply:
Okay let's leave it like you have it for now and we can merge the two later.

Reply:
Sounds good to me. I'll check with @prideout.

Reply:
@bejado In general we should clean up the CMake files after we drop support for clang-in-Visual-Studio and focus on msvc. The logic for Windows builds is getting quite complex.

Reply:
@greg7mdp I'm confused how this is supposed to work. I just tested under Clang and MSVC, and both only print "line one" given the following code:

```
std::cout <<
    R"FILAMENT__(Line one)FILAMENT__",
    R"FILAMENT__(Line two)FILAMENT__",
    R"FILAMENT__(Line three)FILAMENT__";
```

Reply:
Hi @bejado,

The output you show above is from my version of the code, under `if (MSVC_NATIVE)`, right?

In the `else()` case, which should now occur for clang, the  list_licenses function returns one huge string, and the license is generated as follows (here for matc):

```
R"FILAMENT__(License and copyrights for getopt:

Copyright (c) 1987, 1993, 1994, 1995
	The Regents of the University of California.  All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions
are met:
1. Redistribution 

...

ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
------------------------------------------------------------------------------


)FILAMENT__"
```

causing the following error:

```
6>C:\greg\github\filament\out3\tools\matc\generated\licenses/licenses.inc(314,1): error C2026:  string too big, trailing characters truncated
```

I just updated the CMakeLists.txt to merge both versions of the list_licenses function, which should now work fine on all platforms.